### PR TITLE
Create session_latest symlink for Ray sessions

### LINKED ISSUE #5578: Finding the latest session log files is annoying
The log files are timestamped, so if you're running many sessions and want to find the latest logs, you have to manually find the latest timestamp and then `cd` into that directory. Would be much nicer to automatically create a symlink to the latest session so you can `cat /tmp/ray/session_latest/...`.

### PR DESCRIPTION
**Pull request by edoakes**
## What do these changes do?

When Ray starts up, creates or overwrites a symlink at `session_latest` to the latest session directory. If this fails, it will only log a warning, not crash.

## Related issue number

Fixes #5578

## Linter

- [x] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

File: python/ray/utils.py [hunk 1]
```diff
@@ -614,3 +614,48 @@ def try_to_create_directory(directory_path, warn_if_exist=True):
...
+    be created. If a symlink exists in the path, it will be attempted to be
+    removed and replaced.
+
+    Args:
+        directory_path: The path of the existing directory.
+        symlink_path: The path of the symlink to create.
+    """
+    logger = logging.getLogger("ray")
+    directory_path = os.path.expanduser(directory_path)
+    symlink_path = os.path.expanduser(symlink_path)
```
Comment (ericl) [inline python/ray/utils.py]: FYI the canonical way to do this is to create the symlink as a tmp file, and then do an atomic move.

File: python/ray/node.py [hunk 2]
```diff
@@ -19,14 +19,15 @@
...
-from ray.utils import try_to_create_directory
+from ray.utils import try_to_create_directory, try_to_symlink_directory
 
 # Logger for this module. It should be configured at the entry point
 # into the program using Ray. Ray configures it by default automatically
 # using logging.basicConfig in its entry/init points.
 logger = logging.getLogger(__name__)
 
 PY3 = sys.version_info.major >= 3
+SESSION_LATEST = "session_latest"
```
Comment (ericl) [inline python/ray/node.py]: how about just latest?

**Comment by robertnishihara** [comment]
Seems like a great idea, since we're always copying and pasting the session strings...

Do we care about the case where the user simulates multiple nodes on a single machine, e.g., 

```
import ray
from ray.tests.cluster_utils import Cluster

c = Cluster()

c.add_node(include_webui=True)

for _ in range(10):
    c.add_node()

ray.init(redis_address=c.redis_address)
```

In that case, will it just point to the last one out of the 10 created?

cc @suquark who has worked a lot on this code and may have tried something similar in the past.

**Comment by edoakes** [comment]
Yes it will, but I think that's fine. The right way to support that is probably by giving each of them a separate top-level `ray` directory.

**Comment by robertnishihara** [comment]
@edoakes are you still making changes here or should this be merged?

**Comment by edoakes** [comment]
Go ahead and merge.

**Review by robertnishihara — APPROVED** [review]

**Pull request merged**